src/scales/ordinal.js

```diff
@@ -102,14 +102,14 @@ function maybeRound(scale, channels, options) {
   return scale;
 }
 
-function inferDomain(channels) {
+function inferDomain(channels, type) {
   const values = new InternSet();
   for (const {value, domain} of channels) {
     if (domain !== undefined) return domain();
     if (value === undefined) continue;
     for (const v of value) values.add(v);
   }
-  if (values.size > 10e3) throw new Error("This ordinal domain would have more than 10,000 values. If this is intentional, set the domain explicitly.");
+  if (values.size > 10e3 && ["point", "band"].includes(type)) throw new Error("This ordinal domain would have more than 10,000 values. If this is intentional, set the domain explicitly.");
```

Comment:
You can still have a strictly ordinal scale (type = `"ordinal"`) on a position dimension, though it’s usage is rare since you need to specify the range values explicitly. And if you did, I think we probably would want an error for that here, too, so I prefer this instead:


 `registry.get(key) === position` here.
```suggestion
  if (values.size > 10e3 && registry.get(key) === position) throw new Error("This ordinal domain would have more than 10,000 values. If this is intentional, set the domain explicitly.");
```